File: frontend/src/types/category.ts
export interface Category {
  id: number;
  name: string;
  amount: number;
  parent_id: number | null;
  level: number;
  sibling_order: number;
  year_id: number;
}

export interface CategoryTree {
  id: number;
  name: string;
  amount: number;
  parent_id: number | null;
  level: number;
  sibling_order: number;
  year_id: number;
  children: CategoryTree[];
}

export interface CategoryCreate {
  year_id: number;
  name: string;
  amount: number;
  sibling_order: number;
  parent_id: number | null;
}

export interface CategoryUpdate {
  name: string;
  amount: number;
  sibling_order: number;
  parent_id: number | null;
  year_id: number;
}

File: app/models/category.py
from pydantic import BaseModel

class CategoryRead(BaseModel):
    id: int
    name: str
    level: int
    sibling_order: int
    amount: int
    parent_id: int | None = None
    year_id: int

class CategoryTreeRead(BaseModel):
    id: int
    name: str
    level: int
    sibling_order: int
    amount: int
    parent_id: int | None = None
    year_id: int
    children: list["CategoryTreeRead"] | None = None

class CategoryCreate(BaseModel):
    name: str
    amount: int
    sibling_order: int
    parent_id: int | None = None
    year_id: int

class CategoryUpdate(BaseModel):
    name: str
    amount: int
    sibling_order: int
    parent_id: int | None = None

class CategoryOrderUpdate(BaseModel):
    sibling_order: int

# class CategoryTreeCreate(BaseModel):
#     name: str
#     amount: int
#     parent_id: int | None = None
#     fiscal_term_id: int
#     children: list["CategoryTreeCreate"] | None = None

class BudgetBulkUpdateItem(BaseModel):
    id: int
    amount: int

class BudgetBulkUpdate(BaseModel):
    items: list[BudgetBulkUpdateItem]

